src/CommandingOfficers/CommanderInfo.java

```diff
@@ -1,15 +1,42 @@
 package CommandingOfficers;
 
 import java.io.Serializable;
+import java.util.ArrayList;
 
-public class CommanderInfo implements Serializable
+public abstract class CommanderInfo implements Serializable
 {
   public final String name;
-  public final COMaker maker;
+  public ArrayList<InfoPage> infoPages;
 
-  public CommanderInfo(String name, COMaker whichCo)
+  public CommanderInfo(String name)
   {
     this.name = name;
-    maker = whichCo;
+    infoPages = new ArrayList<InfoPage>();
+  }
+
+  public abstract Commander create();
+
+  public static class InfoPage implements Serializable
```

Comment:
Curious: Why does `InfoPage` have to implement `Serializable`?  I would  assume it could just be re-created on load or whatever, instead of needing to read from disk again since they are all static.

Reply:
It's *also* stored by each instance of `Commander`. I could make it `transient`, but then I'd need to add logic to recall that and that seems like too much work.

Reply:
Gotcha. Laziness FTW. As long as our save-file sizes don't get out of hand. Which they wouldn't, from this.